crates/hir-ty/src/tests/traits.rs

```diff
@@ -3955,15 +3935,16 @@ mod module {
 }
 
 fn f() {
+    struct Foo;
     use module::T;
-    impl T for usize {
+    impl T for Foo {
```

Comment:
This is another important (breaking) change. Previously, when checking whether a type impls a trait, we considered three blocks: the type's, the trait's, and the current one. I removed the current one, since *where* we're trait-solving from should have little impact to its result. The `non_local_definitions` lint also warns against this.